yb-voyager/src/migassessment/sizing.go

```diff
@@ -1644,3 +1713,58 @@ func findClosestVersion(targetDbVersion *ybversion.YBVersion, availableVersions
 		return closest.versionId
 	}
 }
+
+/*
+findNumNodesThroughputScalingImportTimeDivisor calculates the relative import time divisor based on throughput scaling with number of nodes.
+
+This function computes a scaling factor to adjust import time estimates based on the number of nodes in the cluster.
+It uses experimental data that shows how import throughput scales with cluster size to determine the appropriate
+time adjustment factor.
+
+Algorithm:
+1. Find the 3-node baseline entry from experimental data (used as reference point)
+2. Find the closest experimental data point to the target number of nodes
+3. Calculate per-node throughput from the closest data point
+4. Estimate total throughput for target nodes: targetNodes * perNodeThroughput
+5. Return ratio: estimatedThroughput / baselineThroughput
+
+This ratio represents how much faster (>1.0) the import would be compared to the 3-node baseline.
+
+Parameters:
+  - numNodesImpactData: slice of experimental data for throughput scaling with different number of nodes
+  - targetNumNodes: the target number of nodes to calculate the scaling ratio for
+
+Returns:
+  - returns the relative import time divisor based on throughput scaling with number of nodes
+*/
+func findNumNodesThroughputScalingImportTimeDivisor(numNodesImpactData []ExpDataLoadTimeNumNodesImpact, targetNumNodes float64) float64 {
+	// select the 3-node baseline entry
+	var baselineData ExpDataLoadTimeNumNodesImpact
+	for _, data := range numNodesImpactData {
+		if data.numNodes.Valid && data.numNodes.Int64 == 3 && data.importThroughputMbps.Valid {
+			baselineData = data
+			break
+		}
+	}
+
+	// Find the closest entry to targetNumNodes
+	closest := numNodesImpactData[0]
```

Comment:
can we do a check in the beginning of this function to check that numNodesImpactData is non-empty. Otherwise this will panic with "index out of range"

Reply:
the data is guaranteed to be present in the experimental database table. But, it is a good check to have, so added it in ```findNumNodesThroughputScalingImportTimeDivisor```